### PR TITLE
[fix]Core dump in SegmentIterator when there is no predicate

### LINKED ISSUE #10453: [Bug] Core dump on test_csv_with_header
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

introduced by #10420
```sql
select count(*) from test_csv_with_header
```

```cpp
start time: 2022年 06月 27日 星期一 15:28:51 CST
*** Aborted at 1656315154 (unix time) try "date -d @1656315154" if you are using GNU date ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 64761 (TID 0x7f7c956cc700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/common/signal_handler.h:407
 1# 0x00007F7D04861920 in /lib64/libc.so.6
 2# doris::segment_v2::SegmentIterator::_init_current_block(doris::vectorized::Block*, std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>, std::allocator<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> > >&) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/olap/rowset/segment_v2/segment_iterator.cpp:825
 3# doris::segment_v2::SegmentIterator::next_batch(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/olap/rowset/segment_v2/segment_iterator.cpp:1031
 4# doris::BetaRowsetReader::next_block(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/olap/rowset/beta_rowset_reader.cpp:189
 5# doris::vectorized::VCollectIterator::Level0Iterator::next(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/vcollect_iterator.cpp:234
 6# doris::vectorized::VCollectIterator::Level1Iterator::_normal_next(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/vcollect_iterator.cpp:425
 7# doris::vectorized::VCollectIterator::Level1Iterator::next(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/vcollect_iterator.cpp:296
 8# doris::vectorized::VCollectIterator::next(doris::vectorized::Block*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/vcollect_iterator.cpp:173
 9# doris::vectorized::BlockReader::_direct_next_block(doris::vectorized::Block*, doris::MemPool*, doris::ObjectPool*, bool*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/block_reader.cpp:160
10# doris::vectorized::BlockReader::next_block_with_aggregation(doris::vectorized::Block*, doris::MemPool*, doris::ObjectPool*, bool*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/olap/block_reader.h:46
11# doris::vectorized::VOlapScanner::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/exec/volap_scanner.cpp:289
12# doris::vectorized::VOlapScanNode::scanner_thread(doris::vectorized::VOlapScanner*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/exec/volap_scan_node.cpp:504
13# void std::__invoke_impl<void, void (doris::vectorized::VOlapScanNode::*&)(doris::vectorized::VOlapScanner*), doris::vectorized::VOlapScanNode*&, doris::vectorized::VOlapScanner*&>(std::__invoke_memfun_deref, void (doris::vectorized::VOlapScanNode::*&)(doris::vectorized::VOlapScanner*), doris::vectorized::VOlapScanNode*&, doris::vectorized::VOlapScanner*&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:74
14# std::__invoke_result<void (doris::vectorized::VOlapScanNode::*&)(doris::vectorized::VOlapScanner*), doris::vectorized::VOlapScanNode*&, doris::vectorized::VOlapScanner*&>::type std::__invoke<void (doris::vectorized::VOlapScanNode::*&)(doris::vectorized::VOlapScanner*), doris::vectorized::VOlapScanNode*&, doris::vectorized::VOlapScanner*&>(void (doris::vectorized::VOlapScanNode::*&)(doris::vectorized::VOlapScanner*), doris::vectorized::VOlapScanNode*&, doris::vectorized::VOlapScanner*&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:97
15# void std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/functional:422
16# void std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>::operator()<, void>() at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/functional:505
17# void std::__invoke_impl<void, std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>&>(std::__invoke_other, std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:61
18# std::enable_if<is_invocable_r_v<void, std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>&>, void>::type std::__invoke_r<void, std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>&>(std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)>&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:117
19# std::_Function_handler<void (), std::_Bind<void (doris::vectorized::VOlapScanNode::*(doris::vectorized::VOlapScanNode*, doris::vectorized::VOlapScanner*))(doris::vectorized::VOlapScanner*)> >::_M_invoke(std::_Any_data const&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/std_function.h:292
20# std::function<void ()>::operator()() const at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/std_function.h:560
21# doris::PriorityWorkStealingThreadPool::work_thread(int) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/util/priority_work_stealing_thread_pool.hpp:135
22# void std::__invoke_impl<void, void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>(std::__invoke_memfun_deref, void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:74
23# std::__invoke_result<void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>::type std::__invoke<void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>(void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:97
24# decltype (__invoke((*this)._M_pmf, (forward<doris::PriorityWorkStealingThreadPool*&>)({parm#1}), (forward<int&>)({parm#1}))) std::_Mem_fn_base<void (doris::PriorityWorkStealingThreadPool::*)(int), true>::operator()<doris::PriorityWorkStealingThreadPool*&, int&>(doris::PriorityWorkStealingThreadPool*&, int&) const at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/functional:131
25# void std::__invoke_impl<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)>&, doris::PriorityWorkStealingThreadPool*&, int&>(std::__invoke_other, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)>&, doris::PriorityWorkStealingThreadPool*&, int&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:61
26# std::enable_if<is_invocable_r_v<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)>&, doris::PriorityWorkStealingThreadPool*&, int&>, void>::type std::__invoke_r<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)>&, doris::PriorityWorkStealingThreadPool*&, int&>(std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)>&, doris::PriorityWorkStealingThreadPool*&, int&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:117
27# void std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/functional:571
28# void std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>::operator()<>() at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/functional:631
29# void std::__invoke_impl<void, std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>>(std::__invoke_other, std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>&&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:61
30# std::__invoke_result<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>>::type std::__invoke<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>>(std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)>&&) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/invoke.h:97
31# void std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)> > >::_M_invoke<0ul>(std::_Index_tuple<0ul>) at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/std_thread.h:253
32# std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)> > >::operator()() at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/std_thread.h:260
33# std::thread::_State_impl<std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPool*, int)> > > >::_M_run() at /home/disk2/pxl/dev/ldb_toolchain/include/c++/11/bits/std_thread.h:211
34# execute_native_thread_routine in /home/disk2/pxl/dev/baidu/bdg/doris/core/output/be/lib/doris_be
35# start_thread in /lib64/libpthread.so.0
36# clone in /lib64/libc.so.6

```

### What You Expected?

fix it

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #10453 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -1007,21 +1010,23 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
...
+                } else if (_is_first_read_column[cid] || i >= block->columns()) {
+                    // if i >= block->columns means the column and not the pred_column means `column i` is
+                    // a delete condition column. but the column is not effective in the segment. so we just
+                    // create a column to hold the data.
+                    // a. origin data -> b. delete condition -> c. new load data
+                    // the segment of c do not effective delete condition, but it still need read the column
+                    // to match the schema.
+                    // TODO: skip read the not effective delete column to speed up segment read.
+                    _current_return_columns[cid] =
+                            Schema::get_data_type_ptr(*column_desc)->create_column();
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1027]: 这个地方是不对的： 对于block 中已经有的column，他原本是要做内存重用的，不会在这生成column

File: be/src/olap/rowset/segment_v2/segment_iterator.h [hunk 2]
```diff
@@ -159,6 +159,7 @@ class SegmentIterator : public RowwiseIterator {
     std::vector<ColumnId>
             _short_cir_pred_column_ids; // keep columnId of columns for short circuit predicate evaluation
     std::vector<bool> _is_first_read_column; // columns hold by segmentIter
+    std::vector<bool> _is_predicate_column;  // columns hold by segmentIter
```
Comment (Gabriel39) [inline be/src/olap/rowset/segment_v2/segment_iterator.h:162]: try not to use `vector<bool>`. Use `vector<char>` or `bitset` instead

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_v2/segment_iterator.h (hunk 2 above)
Comment (mrhhsg) [reply be/src/olap/rowset/segment_v2/segment_iterator.h:162]: This vector will not be accessed frequently, so I think using `vector<bool>` is ok?

**Comment by BiteTheDDDDt** [comment]
```sql
MySQL [test]> select count(*) from test;
+----------+
| count(*) |
+----------+
|    60017 |
+----------+
1 row in set (0.181 sec)

MySQL [test]> select count(*) from test;
+----------+
| count(*) |
+----------+
|    45150 |
+----------+
1 row in set (0.036 sec)
```
Hi, I do test and get wrong result

**Pull request closed without merging**